src/core/geometry/geom3/difference.js

```diff
@@ -20,34 +21,29 @@ const Tree = require('../trees')
    *      |       |
    *      +-------+
    */
-const subtract = (otherGeom3, geometry) => {
-  let geometries
-  if (geometry instanceof Array) {
-    geometries = geometry
-  } else {
-    geometries = [geometry]
-  }
-  let result = otherGeom3
-  for (let i = 0; i < geometries.length; i++) {
-    let islast = (i === (geometries.length - 1))
-    result = subtractSub(result, geometries[i], islast, islast)
-  }
-  return result
-}
-
 const subtractSub = (otherGeom3, geometry, doRetesselate, doCanonicalize) => {
-  let a = new Tree(otherGeom3.polygons)
-  let b = new Tree(geometry.polygons)
+  const a = new Tree(otherGeom3.polygons)
+  const b = new Tree(geometry.polygons)
   a.invert()
   a.clipTo(b)
   b.clipTo(a, true)
   a.addPolygons(b.allPolygons())
   a.invert()
   let result = fromPolygons(a.allPolygons())
-  if (doRetesselate) result = retesselate(result)
   if (doCanonicalize) result = canonicalize(result)
+  if (doRetesselate) result = retessellate(result)
+  return result
+}
+
+const subtract = (otherGeom3, ...geometries) => {
+  let result = otherGeom3
```

Comment:
this will not work in case somebody passes a single item by accident, you need to
- add `toArray` to enforce always getting an array
- thrown an error if array length is 1

Reply:
How about this? Trying to take adavantage of the rest functionality.

```
const subtract = (...geometries) => {
  let result = geometries.shift()
  ...
```

Reply:
Things will blow up if less then two geometries are provided. But checking the length of geometries can also be performed. We should be consistent.

Which way is it? I prefer natural implosion at this level. Of course, the API version of differene() should be checking.

Reply:
Why not consider subtracting no geometries as a no-op?